archimedes/archimedes/functions/plotting/scatter_plot.py

```diff
@@ -88,3 +95,251 @@ def scatter_plotly(
 
     return fig
 
+def scatter_plotnine(
+    data_frame: pd.DataFrame,
```

Comment:
This, function looks, kinda awful lol.  So many arguments, this implies really that there is "some structure" `class ScatterPlotNineOptions` which lets you build the figure based on several options.  It'd allow you to break out this big function into smaller utilities and probably do so code sharing, too.

Right now, if I'm honest, this feels like a mess :P